bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java

```diff
@@ -0,0 +1,474 @@
+import org.eclipse.jetty.websocket.client.ClientUpgradeRequest;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.tibber.internal.config.TibberConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link TibberHandler} is responsible for handling queries to/from Tibber API.
+ *
+ * @author Stian Kjoglum - Initial contribution
+ */
+@NonNullByDefault
+public class TibberHandler extends BaseThingHandler {
+    private static final int REQUEST_TIMEOUT = (int) TimeUnit.SECONDS.toMillis(20);
+    private Logger logger = LoggerFactory.getLogger(TibberHandler.class);
+    private final Properties httpHeader = new Properties();
+    private SslContextFactory sslContextFactory = new SslContextFactory(true);
+    private @Nullable WebSocketClient client;
+    private @Nullable TibberWebSocketListener socket;
+    private @Nullable Session session;
+    private @Nullable ClientUpgradeRequest request;
+    private boolean connected = false;
+    private String rtEnabled = "false";
+
+    private @Nullable TibberConfiguration configuration;
+    private @Nullable TibberPriceConsumptionHandler priceInfo = new TibberPriceConsumptionHandler();
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    public TibberHandler(Thing thing) {
+        super(thing);
+    }
+
+    @SuppressWarnings("null")
+    @Override
+    public void initialize() {
+        logger.info("Initializing Tibber API");
+        updateStatus(ThingStatus.UNKNOWN);
+
+        configuration = getConfigAs(TibberConfiguration.class);
+        getTibberParameters();
+        startRefresh(configuration.getRefresh());
+    }
+
+    @SuppressWarnings("null")
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+        if (command instanceof RefreshType) {
+            startRefresh(configuration.getRefresh());
+
+        } else {
+            logger.debug("Tibber API is read-only and does not handle commands");
+        }
+        return;
```

Comment:
yea, you get the point, please remove any blank return statements at the end of other methods.

Reply:
Corrected.